modules/imgproc/include/opencv2/imgproc/hal/hal.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef CV_IMGPROC_HAL_HPP
+#define CV_IMGPROC_HAL_HPP
+
+#include "opencv2/core/cvdef.h"
+#include "opencv2/core/cvstd.hpp"
+#include "opencv2/core/hal/interface.h"
+
+namespace cv { namespace hal {
+
+//! @addtogroup core_hal_functions
+//! @{
+
+struct CV_EXPORTS Filter2D
```

Comment:
I would actually split it into 2 classes. One reason is clarity. The other reason is theorecitcal possiblity for the linker to throw away more unused code.

createFilter2D/createSepFilter2D can then be renamed to simple "create"